src/pydvl/influence/torch/functional.py

```diff
@@ -180,27 +194,25 @@ def get_hvp_function(
         k: p if track_gradients else p.detach() for k, p in model.named_parameters()
     }
 
-    def hvp_function(vec: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+    def hvp_function(vec: torch.Tensor) -> torch.Tensor:
         v = align_structure(params, vec)
         empirical_loss = empirical_loss_function(model, loss, data_loader)
-        return hvp(empirical_loss, params, v, reverse_only=reverse_only)
+        return flatten_tensors_to_vector(
```

Comment:
Sorry for the naive question: This is probably not the right place nor the only one to make this comment, but there seem to be a few places where lists of tensors are constructed and then concatenated, which would double memory requirements. In this case, given an input tensor and the list of parameters, it should be possible to preallocate the output tensor for _all_ the jvp/vjp operations, right? Maybe the slicing would become a bit tricky, though. I'm not sure where the biggest memory issues arise, but is this case on the radar? Or is it not worth considering?

Reply:
Created an issue #401